kafka-transport-binding.md

```diff
@@ -0,0 +1,281 @@
+
+If a receiver finds a CloudEvents media type as per the above rule, but with an
+event format that it cannot handle, for instance
+`application/cloudevents+avro`, it MAY still treat the event as binary and
+forward it to another party as-is.
+
+### 3.1. Key Attribute
+The 'key' attribute is populated by a partitionKeyExtractor function. The 
+partitionKeyExtractor is a transport specific function that contains bespoke logic 
+to extract and populate the value. A default implementation of the extractor will 
+use the [Partitioning](extensions/partitioning.md) extension value. 
+
+### 3.2. Binary Content Mode
+
+The *binary* content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.2.1. Content Type
+
+For the *binary* mode, the header `ce_contentType` property MUST be
+mapped directly to the CloudEvents `contentType` attribute.
+
+
+#### 3.2.2. Event Data Encoding
+
+The [`data` attribute](#21-data-attribute) byte-sequence MUST be used as the
+value of the Kafka message.
+
+#### 3.2.3. Metadata Headers
+
+All [CloudEvents][CE] attributes and
+[CloudEvent Attributes Extensions](primer.md#cloudevent-attribute-extensions) 
+with exception of `data` MUST be individually mapped to and from the Header 
+fields in the Kafka message.
+
+##### 3.2.3.1 Property Names
+
+CloudEvent attributes are prefixed with "ce_" for use in the
+[message-headers][Kafka-Message-Header] section.
+
+Examples:
+
+    * `time` maps to `ce_time`
+    * `id` maps to `ce_id`
+    * `specversion` maps to `ce_specversion`
+
+##### 3.2.4.2 Property Values
+
+The value for each Kafka header is constructed from the respective
+header's Kafka representation, compliant with the [Kafka message
+format][Kafka-Message-Format] specification.
+
+
+#### 3.2.5 Example
+
+This example shows the *binary* mode mapping of an event into the
+Kafka message. All other CloudEvents attributes
+are mapped to Kafka Header fields with prefix `ce_`.
+
+Mind that `ce_` here does refer to the event `data`
+content carried in the payload.
+
+``` text
+------------------ Message -------------------
+
+Topic Name: mytopic
+
+------------------- key ----------------------
+
+Key: mykey
+
+------------------ headers -------------------
+
+ce_specversion: "0.4-wip"
+ce_type: "com.example.someevent"
+ce_source: "/mycontext/subcontext"
+ce_id: "1234-1234-1234"
+ce_time: "2018-04-05T03:56:24Z"
+ce_datacontenttype: application/avro
+       .... further attributes ...
+
+------------------- value --------------------
+
+            ... application data ...
+
+-----------------------------------------------
+```
+
+### 3.3. Structured Content Mode
+
+The *structured* content mode keeps event metadata and data together in the
+payload, allowing simple forwarding of the same event across multiple routing
+hops, and across multiple transports.
+
+#### 3.3.1. Kafka Content-Type
+
+The [Kafka `ce_contentType`] property field MUST be set to the media
```

Comment:
"T" -> "t"

Reply:
Actually, I think the "ce_" here is incorrect since this isn't a CE property. Right?

Reply:
agree